subcommand/inject-connect/command.go

```diff
@@ -162,6 +167,12 @@ func (c *Command) init() {
 		"Overwrite Kubernetes probes to point to Envoy by default when in Transparent Proxy mode.")
 	c.flagSet.BoolVar(&c.flagEnableOpenShift, "enable-openshift", false,
 		"Indicates that the command runs in an OpenShift cluster.")
+	c.flagSet.StringVar(&c.flagConsulClientPortHTTP, "consul-client-port-http", "8500",
+		"Consul client HTTP port, defaults to 8500.")
+	c.flagSet.StringVar(&c.flagConsulClientPortHTTPS, "consul-client-port-https", "8501",
+		"Consul client HTTPS port, defaults to 8501.")
```

Comment:
You might want to re-use the fact that we're setting CONSUL_HTTP_ADDR as env variable for the deployment and parsing the URL to get the port here: https://github.com/hashicorp/consul-k8s/blob/d49e91985755fe71b123c933619bc26a4c0e16ec/subcommand/inject-connect/command.go#L313-L327